external/loaders/loaders/batches/_batch.py

```diff
@@ -163,8 +158,13 @@ def batches_from_mapper(
     if unstacked_dims is not None:
         transforms.append(curry(stack)(unstacked_dims))
         transforms.append(shuffle)
```

Comment:
I think in the future we'd want an option to sample over a dimension like time, x/y/tile, or dataset (for Spencer) evenly eg using `xr.Dataset.thin` and then stack+shuffle _after_ sampling. I'm thinking about this since at C384 the subsampling is probably going to have to be severe and the odds of getting batches that are unbalanced in some way seem higher. Is this API suitable to adding that option later? `subsample_ratio` might be so specific that it kind of precludes that approach. I guess we could add a `subsample_every_n` arg that's mutually exclusive with `subsample_ratio`. Something to keep in mind.

Reply:
Following our standup discussion, I'll leave this as-is for now. It should be fairly easy to update this later to swap out the shuffling transform based on an argument, or just replace it completely.

Reply:
Sounds good